### PR TITLE
chore(docs): Algolia search guide from class to functional components

### PR DESCRIPTION
**Pull request by janosh**
## Description

This PR updates the Algolia search guide from class to functional components using React Hooks.

### CODE REVIEW

File: docs/docs/adding-search-with-algolia.md [hunk 1]
```diff
@@ -2,33 +2,39 @@
...
+First, you'll need to add [`gatsby-plugin-algolia`](https://github.com/algolia/gatsby-plugin-algolia) and [`react-instantsearch-dom`](https://github.com/algolia/react-instantsearch) to your project. `react-instantsearch` is Algolia's library containing off-the-shelf React components which we can import to save ourselves a lot of work. If you're not using it already, also install [`dotenv`](https://github.com/motdotla/dotenv) while you're at it. We're going to need it to specify your Algolia app ID and both the search and admin API keys without commiting them to version control.
 
-```shell
+```sh
```
Comment (LekoArts) [inline docs/docs/adding-search-with-algolia.md]: Please keep `shell`

```suggestion
```shell
```
https://www.gatsbyjs.org/contributing/gatsby-style-guide/#format-code-blocks-inline-code-and-images

File: docs/docs/adding-search-with-algolia.md [hunk 2]
```diff
@@ -2,33 +2,39 @@
...
+First, you'll need to add [`gatsby-plugin-algolia`](https://github.com/algolia/gatsby-plugin-algolia) and [`react-instantsearch-dom`](https://github.com/algolia/react-instantsearch) to your project. `react-instantsearch` is Algolia's library containing off-the-shelf React components which we can import to save ourselves a lot of work. If you're not using it already, also install [`dotenv`](https://github.com/motdotla/dotenv) while you're at it. We're going to need it to specify your Algolia app ID and both the search and admin API keys without commiting them to version control.
```
Comment (LekoArts) [inline docs/docs/adding-search-with-algolia.md]:
```suggestion
First, you'll need to add [`gatsby-plugin-algolia`](https://github.com/algolia/gatsby-plugin-algolia) and [`react-instantsearch-dom`](https://github.com/algolia/react-instantsearch) to your project. `react-instantsearch` is Algolia's library containing off-the-shelf React components which you can import to save ourselves a lot of work. You'll also be using `dotenv` which gets shipped with Gatsby by default. You're going to need it to specify your Algolia app ID and both the search and admin API keys without commiting them to version control.
```

dotenv already comes with Gatsby. Also: https://www.gatsbyjs.org/contributing/gatsby-style-guide/#use-you-as-the-pronoun

File: docs/docs/adding-search-with-algolia.md [hunk 3]
```diff
@@ -2,33 +2,39 @@
...
 
-```shell
+```sh
 yarn add gatsby-plugin-algolia react-instantsearch-dom algoliasearch dotenv
```
Comment (LekoArts) [inline docs/docs/adding-search-with-algolia.md]:
```suggestion
npm install --save gatsby-plugin-algolia react-instantsearch-dom algoliasearch
```

File: docs/docs/adding-search-with-algolia.md [hunk 4]
```diff
@@ -40,32 +46,33 @@ module.exports = {
...
     },
+    `gatsby-plugin-styled-components`,
   ],
 }
 ```
 
 Notice that we're loading `queries` from a file at `./src/utils/algolia.js` (you can of course put it wherever you like) and our Algolia ID and API key from `.env` so let's add those files.
 
-```env:title=.env
+```sh:title=.env
```
Comment (LekoArts) [inline docs/docs/adding-search-with-algolia.md]:
```suggestion
```text:title=.env
```

File: docs/docs/adding-search-with-algolia.md [hunk 5]
```diff
@@ -98,7 +105,7 @@ const postQuery = `{
         frontmatter {
           title
           slug
-          date(formatString: "MMM DD, YYYY")
+          date(formatString: "MMM D, YYYY")
```
Comment (LekoArts) [inline docs/docs/adding-search-with-algolia.md]: Why this change?

File: docs/docs/adding-search-with-algolia.md [hunk 6]
```diff
@@ -2,33 +2,39 @@
...
+First, you'll need to add [`gatsby-plugin-algolia`](https://github.com/algolia/gatsby-plugin-algolia) and [`react-instantsearch-dom`](https://github.com/algolia/react-instantsearch) to your project. `react-instantsearch` is Algolia's library containing off-the-shelf React components which we can import to save ourselves a lot of work. If you're not using it already, also install [`dotenv`](https://github.com/motdotla/dotenv) while you're at it. We're going to need it to specify your Algolia app ID and both the search and admin API keys without commiting them to version control.
 
-```shell
+```sh
 yarn add gatsby-plugin-algolia react-instantsearch-dom algoliasearch dotenv
 ```
 
-Next, add `gatsby-plugin-algolia` to your `gatsby-config.js`.
+We will be using `styled-components` to design the search UI in this guide but you can use whichever CSS solution you prefer. If you'd like to start using `styled-components` as well, you also need to install
```
Comment (LekoArts) [inline docs/docs/adding-search-with-algolia.md]:
```suggestion
You will be using `styled-components` to design the search UI in this guide but you can use whichever CSS solution you prefer. If you'd like to start using `styled-components` as well, you also need to install
```

File: docs/docs/adding-search-with-algolia.md [hunk 7]
```diff
@@ -2,33 +2,39 @@
...
+
+```sh
```
Comment (LekoArts) [inline docs/docs/adding-search-with-algolia.md]:
```suggestion
```shell
```

File: docs/docs/adding-search-with-algolia.md [hunk 8]
```diff
@@ -2,33 +2,39 @@
...
+yarn add react styled-components gatsby-plugin-styled-components
```
Comment (LekoArts) [inline docs/docs/adding-search-with-algolia.md]:
```suggestion
npm install --save styled-components gatsby-plugin-styled-components
```

File: docs/docs/adding-search-with-algolia.md [hunk 9]
```diff
@@ -2,33 +2,39 @@
...
+```
+
+Next, add `gatsby-plugin-algolia` and `gatsby-plugin-styled-components` if you installed it to your `gatsby-config.js`.
```
Comment (LekoArts) [inline docs/docs/adding-search-with-algolia.md]:
```suggestion
Next, add `gatsby-plugin-algolia` and `gatsby-plugin-styled-components` to your `gatsby-config.js`.
```

File: docs/docs/adding-search-with-algolia.md [hunk 10]
```diff
@@ -132,152 +139,197 @@ const queries = [
...
+- **`index.js`**: the main component
+- **`input.js`**: the text input field
+- **`hitComps.js`**: the components that will render matching posts/pages
+- **`styles.js`**: the styled components
+
+There's quite a lot happening in these files so let's break them down one by one and piece by piece.
+
+### `index.js`
+
+```jsx:title=src/components/search/index.js
```
Comment (LekoArts) [inline docs/docs/adding-search-with-algolia.md]:
```suggestion
```js:title=src/components/search/index.js
```

File: docs/docs/adding-search-with-algolia.md [hunk 11]
```diff
@@ -297,74 +349,17 @@ export default connectSearchBox(({ refine, ...rest }) => (
...
-  root={{ Root, props: { ref } }}
->
-```
+Now let's look at the styled components `SearchIcon`, `Form`, `Input` as well as the ones imported in `index.js`.
 
-Now we come to the styled components. Of course, the design will be very different from one site to the next so we only list them here for completeness and because it might save some time to simply copy and customize them.
+## `styled.js`
 
-```js:title=src/components/Search/styles.js
+```jsx:title=src/components/search/styles.js
```
Comment (LekoArts) [inline docs/docs/adding-search-with-algolia.md]:
```suggestion
```js:title=src/components/search/styles.js
```

File: docs/docs/adding-search-with-algolia.md (hunk 2 above)
Comment (DSchau) [reply docs/docs/adding-search-with-algolia.md]: @LekoArts it's a bit non-ideal to rely on downstream dependencies to exist (and be the expected version), so I'd keep the installation of dotenv.

Comment (LekoArts) [reply docs/docs/adding-search-with-algolia.md]: mh, interesting. We do the same (relying on that fact) everywhere else I think, e.g. in the env variables documentation.

Comment (janosh) [reply docs/docs/adding-search-with-algolia.md]: I'm happy to go either way but I agree with @DSchau that it's often good to keep dependencies explicit. Of course, consistency is nice as well.

File: docs/docs/adding-search-with-algolia.md (hunk 4 above)
Comment (janosh) [reply docs/docs/adding-search-with-algolia.md]: I usually get better syntax highlighting (e.g. of comments) in `.env` files with `sh` but I assume `text` is convention?

File: docs/docs/adding-search-with-algolia.md (hunk 5 above)
Comment (janosh) [reply docs/docs/adding-search-with-algolia.md]: Looks better in my opinion. Just a small oversight during the original writing of this guide but we can also keep it as is.

Comment (LekoArts) [reply docs/docs/adding-search-with-algolia.md]: Okay 👍

**Comment by janosh** [comment]
@LekoArts Thanks a lot for making the change from "we" to "you". Sorry for the slow reaction. Was travelling with poor connection these past couple of days but I'm back now and just changed all `jsx` code blocks to `js` and added titles where they were missing.

**Comment by LekoArts** [comment]
@janosh No need to apologize, thank you very much for your effort! :)
LGTM 👍
